src/m2n/PointToPointCommunication.cpp

```diff
@@ -401,6 +401,91 @@ void PointToPointCommunication::acceptConnection(std::string const &acceptorName
   _isConnected = true;
 }
 
+void PointToPointCommunication::acceptPreConnection(std::string const &acceptorName,
+                                                 std::string const &requesterName)
+{
+  TRACE(acceptorName, requesterName);
+  assertion(not isConnected(), "Already connected!");
+  CHECK(utils::MasterSlave::_masterMode || utils::MasterSlave::_slaveMode,
+        "You can only use a point-to-point communication between two participants which both use a master. "
+            << "Please use distribution-type gather-scatter instead.");
+
+  if (utils::MasterSlave::_masterMode) {
+    // Establish connection between participants' master processes.
+    auto c = _communicationFactory->newCommunication();
+
+    c->acceptConnection(acceptorName, requesterName, utils::MasterSlave::_rank);
+
+    int requesterMasterRank;
+
+    // Exchange ranks of participants' master processes.
+    c->send(utils::MasterSlave::_masterRank, 0);
+    c->receive(requesterMasterRank, 0);
+
+  } else {
+    assertion(utils::MasterSlave::_slaveMode);
+  }
+
+  std::vector<int> localConnectedRanks = _mesh->getConnectedRanks();
+
+  if (localConnectedRanks.empty()) {
+    _isConnected = true;
```

Comment:
Since the actual connections are created here, I suggest we keep it as it is.

Reply:
True.

Reply:
Could you add a test which covers this case?

Reply:
I have no idea how to test this case, because we test the connection by exchanging a double. If there is no connection, whit whom this rank should exchange the data?